### PR TITLE
Add check artifact hash action

### PR DESCRIPTION
**Pull request by domire8**
## Description

This PR adds an action that checks if an artifact that contains the latest hash of a directory exists. It can be reused in dynamic components (https://github.com/aica-technology/dynamic-components/pull/302) and hardware collections (https://github.com/aica-technology/hardware-collections/pull/74) to check if a component/collection needs rebuilding or not.

## Review guidelines

Estimated Time of Review: 3 minutes

### Blocks:
- https://github.com/aica-technology/hardware-collections/pull/74
- https://github.com/aica-technology/dynamic-components/pull/302

### CODE REVIEW

File: .github/actions/check-artifact-hash/action.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  steps:
+    - id: parse
+      name: Parse artifact name
+      run: |
+        COMP_PATH=${{ inputs.package_path }}
+        COMP_PATH=${COMP_PATH//\//_}
+        echo "artifact_name=${COMP_PATH:2}" >> $GITHUB_OUTPUT
+      shell: bash
+
+    - uses: dawidd6/action-download-artifact@v2
```
Comment (domire8) [inline .github/actions/check-artifact-hash/action.yml:31]: This action checks the generated artifacts of the latest successful workflow (build-release.yaml in our case) on a specific branch and downloads the artifact with name `name` if available

File: .github/actions/check-artifact-hash/action.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      with:
+        workflow: build-release.yaml
+        workflow_conclusion: success
+        branch: ${{ github.ref_name }}
```
Comment (LouisBrunner) [inline .github/actions/check-artifact-hash/action.yml:35]: Why restrict to the branch? Without we could use this to build on feature branches with minimal rebuild

**Review by LouisBrunner — COMMENTED** [review]
Let's add `save-artifact-hash` too? It's not a lot of lines of code but it's definitely not something we should copy paste across repos

**Comment by domire8** [comment]
> Let's add `save-artifact-hash` too? It's not a lot of lines of code but it's definitely not something we should copy paste across repos

What do we gain by doing that? It is already an action?

File: .github/actions/check-artifact-hash/action.yml (hunk 2 above)
Comment (domire8) [reply .github/actions/check-artifact-hash/action.yml:35]: You mean that if we open a PR on a feature branch we could check the diff to the target branch? not a bad idea :+1: Should I make this branch an input to the action?

Comment (LouisBrunner) [reply .github/actions/check-artifact-hash/action.yml:35]: Basically, but I think just using the target branch opens other issue. Really we should have a way to track multiple hashes 😕

Comment (domire8) [reply .github/actions/check-artifact-hash/action.yml:35]: Given that for now HC and dynamic-components only run that workflow on develop and main, this is good to go for now no? I see the advantages of multiple hashes for the future though

**Review by LouisBrunner — APPROVED** [review]

**Pull request merged**